File: backend/router/web/knowledge_base.py
from common.base_router import NO_AUTH, ApiRouter
from common.entity.base_response import BaseResponse
from common.entity.schemas.knowledge_base import (
    KnowledgeBaseCreateRequest,
    KnowledgeBaseListRequest,
    KnowledgeBaseListResponse,
    KnowledgeBaseResponse,
    KnowledgeBaseUpdateRequest,
)
from services import KnowledgeBaseService

# 默认所有路由需要认证
router = ApiRouter(prefix="/api/v1/knowledge-base", tags=["knowledge-base"])


@router.post("/list", response_model=BaseResponse[KnowledgeBaseListResponse])
async def get_kb_list(request: KnowledgeBaseListRequest):
    """
    获取知识库列表

    - 需要认证
    - 支持分页
    """
    result = await KnowledgeBaseService.get_kb_list(
        page=request.page, page_size=request.pageSize
    )
    return BaseResponse(code=0, message="获取知识库列表成功", data=result)


@router.post("/get", response_model=BaseResponse[KnowledgeBaseResponse])
async def get_kb(id: int):
    """
    获取单个知识库

    - 需要认证
    - 参数: id (query parameter)
    """
    result = await KnowledgeBaseService.get_kb(id)
    return BaseResponse(code=0, message="获取知识库成功", data=result)


@router.post("/create", response_model=BaseResponse[KnowledgeBaseResponse])
async def create_kb(request: KnowledgeBaseCreateRequest):
    """
    创建知识库

    - 需要认证
    - 知识库名称不能为空
    """
    result = await KnowledgeBaseService.create_kb(request)
    return BaseResponse(code=0, message="创建知识库成功", data=result)


@router.post("/update", response_model=BaseResponse[KnowledgeBaseResponse])
async def update_kb(id: int, request: KnowledgeBaseUpdateRequest):
    """
    更新知识库

    - 需要认证
    - 参数: id (query parameter)
    """
    result = await KnowledgeBaseService.update_kb(id, request)
    return BaseResponse(code=0, message="更新知识库成功", data=result)


@router.post("/delete", response_model=BaseResponse[dict])
async def delete_kb(id: int):
    """
    删除知识库

    - 需要认证
    - 参数: id (query parameter)
    """
    await KnowledgeBaseService.delete_kb(id)
    return BaseResponse(code=0, message="删除知识库成功", data={"deleted": True})

File: front/src/lib/api/knowledgeBase.ts
import { apiClient } from './client';
import { mockKnowledgeBases } from './mock';
import type { BaseResponse, PaginatedResponse, KnowledgeBase, CreateKnowledgeBaseRequest } from '../types';

const USE_MOCK_API = process.env.NEXT_PUBLIC_USE_MOCK_API === 'true';

export const knowledgeBaseApi = {
  /**
   * List knowledge bases
   */
  list: async (params: { page?: number; pageSize?: number } = {}) => {
    if (USE_MOCK_API) {
      const { page = 1, pageSize = 10 } = params;
      const start = (page - 1) * pageSize;
      const end = start + pageSize;
      const items = mockKnowledgeBases.slice(start, end);

      return Promise.resolve({
        code: 0,
        message: 'success',
        data: {
          total: mockKnowledgeBases.length,
          items,
        },
      } as BaseResponse<PaginatedResponse<KnowledgeBase>>);
    }

    return apiClient.post<BaseResponse<PaginatedResponse<KnowledgeBase>>>(
      '/api/v1/knowledge-base/list',
      params
    );
  },

  /**
   * Get knowledge base by ID (using query parameter)
   */
  getById: async (id: number) => {
    if (USE_MOCK_API) {
      const kb = mockKnowledgeBases.find((k) => k.id === id);
      if (!kb) {
        return Promise.reject({ response: { status: 404 } });
      }
      return Promise.resolve({
        code: 0,
        message: 'success',
        data: kb,
      } as BaseResponse<KnowledgeBase>);
    }

    return apiClient.post<BaseResponse<KnowledgeBase>>('/api/v1/knowledge-base/get', null, {
      params: { id },
    });
  },

  /**
   * Create knowledge base
   */
  create: async (data: CreateKnowledgeBaseRequest) => {
    if (USE_MOCK_API) {
      const newKB: KnowledgeBase = {
        id: Date.now(),
        ...data,
        document_count: 0,
        created_at: new Date().toISOString(),
        updated_at: new Date().toISOString(),
      };
      mockKnowledgeBases.unshift(newKB);

      return Promise.resolve({
        code: 0,
        message: 'success',
        data: newKB,
      } as BaseResponse<KnowledgeBase>);
    }

    return apiClient.post<BaseResponse<KnowledgeBase>>('/api/v1/knowledge-base/create', data);
  },

  /**
   * Update knowledge base (using query parameter)
   */
  update: async (id: number, data: Partial<CreateKnowledgeBaseRequest>) => {
    if (USE_MOCK_API) {
      const kb = mockKnowledgeBases.find((k) => k.id === id);
      if (!kb) {
        return Promise.reject({ response: { status: 404 } });
      }
      Object.assign(kb, data, { updated_at: new Date().toISOString() });

      return Promise.resolve({
        code: 0,
        message: 'success',
        data: kb,
      } as BaseResponse<KnowledgeBase>);
    }

    return apiClient.post<BaseResponse<KnowledgeBase>>('/api/v1/knowledge-base/update', data, {
      params: { id },
    });
  },

  /**
   * Delete knowledge base (using query parameter)
   */
  delete: async (id: number) => {
    if (USE_MOCK_API) {
      const index = mockKnowledgeBases.findIndex((k) => k.id === id);
      if (index === -1) {
        return Promise.reject({ response: { status: 404 } });
      }
      mockKnowledgeBases.splice(index, 1);

      return Promise.resolve({
        code: 0,
        message: 'success',
        data: { deleted: true },
      } as BaseResponse<{ deleted: boolean }>);
    }

    return apiClient.post<BaseResponse<{ deleted: boolean }>>('/api/v1/knowledge-base/delete', null, {
      params: { id },
    });
  },
};
